include/alpaka/test/event/EventHostManualTrigger.hpp

```diff
@@ -57,6 +57,8 @@ namespace alpaka::test
                     m_bIsReady = true;
                 }
                 m_conditionVariable.notify_one();
+                // Give alpaka time to update into the new state, process all events and tasks.
+                std::this_thread::sleep_for(std::chrono::milliseconds(200u));
```

Comment:
I am very concerned if any of our multithreaded logic depends on a wait. Is there no other way to establish a synchronization?

Reply:
I do not know any other way because we do not know how long CUDA or alpaka needs to run async tasks.
Calling wait or any other function is also not good because if you build your runtime engine as MPI is doing it you will always perform actions if an API call to the library is performed instead of asynchronous in the background.

IMO sleeping here is not 100% safe but OKisch. It is not affecting the core functionality of alpaka since this is a class used only for testing.

Reply:
Could we wait on an atomic variable here?

Reply:
I agree, that it is OKish because it is part of the unit tests. I assume we may see the test fail sometimes, but we can always restart the CI and cross fingers. So I think we can accept this change and fix it in the future if the CI fails to often.

Reply:
> Could we wait on an atomic variable here?

No because we can not influence what CUDA7HIP is doing in the background and even in alpaka we can to track the state.
Please keep in mind that this class is explicitly made to emulate the asynchronous behaviors of a threaded system. We control when a kernel is ready to have a well-defined state in which events and kernels must be executed.

Reply:
> I agree, that it is OKish because it is part of the unit tests. I assume we may see the test fail sometimes, but we can always restart the CI and cross fingers. So I think we can accept this change and fix it in the future if the CI fails to often.

I do not think we will ever see it failing because of the sleep. 200ms are ages compared to what we do within alpaka and even CUDA will never be so slow that we will hit the border.